File: integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/Address.java
package io.quarkus.it.panache;

import jakarta.persistence.Entity;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

@Entity
public class Address extends PanacheEntity implements Comparable<Address> {

    public String street;

    public Address() {
    }

    public Address(String street) {
        this.street = street;
    }

    @Override
    public int compareTo(Address address) {
        return street.compareTo(address.street);
    }
}

File: integration-tests/hibernate-orm-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/kotlin/Address.kt
package io.quarkus.it.panache.kotlin

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanionBase
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id

@Entity
open class Address : PanacheEntityBase {
    companion object : PanacheCompanionBase<Address, Int> {
        override fun count(query: String, params: Map<String, Any>): Long {
            AddressDao.shouldBeOverridden()
        }
    }

    @Id
    @GeneratedValue
    @JvmField
    var id: Int? = null

    var street: String? = null

    var street2: String? = null

    constructor()

    constructor(street: String) {
        this.street = street
    }

    override fun toString() = "${javaClass.simpleName}<$id>"
}
